engine/pong.cpp

```diff
@@ -62,13 +62,14 @@ void Pong::Init()
 
 void Pong::GameLoop()
 {
+    GetInstance().mTimer.HandleTimerCallbacks();
```

Comment:
I would possibly be encouraged to use the fully qualified version here, aka `Pong::GetInstance()` just for clarity.

I may also add a getter to `GetTimer` instead of interacting with the raw member variable (same with collision manager)